src/service/resources-impl.js

```diff
@@ -1340,6 +1354,14 @@ export class Resource {
      * @private {number|undefined}
      */
     this.pendingChangeHeight_;
+
+    /** @private {?function(*)} */
+    this.whenFirstLayoutCompleteResolve_ = null;
+
+    /** @private @const {!Promise} */
+    this.whenFirstLayoutCompletePromise_ = new Promise(resolve => {
```

Comment:
I'd rather avoid an extra promise here. Let's defer creation until it's requested.